parsl/probe/probe.py

```diff
@@ -0,0 +1,273 @@
+        for i, executor in enumerate(executors):
+            logger.debug(f'\tBegin checking #{i+1} executor in the config')
+            executor.provider.script_dir = '.'
+            executor.provider.channel.script_dir = '.'
+            block_id = executor.start()
+            spec = None
+            if type(executor) is WorkQueueExecutor:
+                spec = avail_spec
+            fut = executor.submit(func, spec, arg)
+            try:
+                if fut.result() == expected:
+                    logger.debug(f'\tWhen starting block {block_id}, #{i+1} executor worked properly')
+                else:
+                    logger.error(f'\tWhen starting block {block_id}, #{i+1} executor did NOT work properly')
+            except Exception:
+                raise
+
+
+def start_logger(filename, name='parsl', level=logging.DEBUG, format_string=None):
+    """Add a file handler and a stream handler.
+
+    Args:
+        - filename (string): Name of the file to write logs to
+        - name (string): Logger name
+        - level (logging.LEVEL): Set the logging level.
+        - format_string (string): Set the format string
+
+    Returns:
+       -  None
+    """
+    if format_string is None:
+        format_string = "%(asctime)s.%(msecs)03d %(name)s:%(lineno)d [%(levelname)s]  %(message)s"
+    formatter = logging.Formatter(format_string, datefmt='%Y-%m-%d %H:%M:%S')
+
+    global logger
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.DEBUG)
+
+    file_handler = logging.FileHandler(filename)
+    file_handler.setLevel(level)
+    file_handler.setFormatter(formatter)
+    logger.addHandler(file_handler)
+
+    stream_handler = logging.StreamHandler()
+    stream_handler.setLevel(level)
+    stream_handler.setFormatter(formatter)
+    logger.addHandler(stream_handler)
+
+
+def cli_run():
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f", "--file", required=True,
+                        help="Config file to be tested")
+    parser.add_argument("-d", "--debug", action='store_true',
+                        help="Set logging debug mode")
+    parser.add_argument("-l", "--logdir", default="config_probe_logs",
+                        help="Config probe log directory")
+    args = parser.parse_args()
+
+    try:
+        os.makedirs(args.logdir, exist_ok=True)
+        probe_log_file = os.path.join(args.logdir, 'parsl-probe.log')
+        start_logger(probe_log_file,
+                     name='probe_log',
+                     level=logging.DEBUG if args.debug else logging.INFO)
+
+        src = SourceFileLoader("config", args.file).load_module()
+        config = src.config
+        config_tester = ConfigTester(config)
+    except Exception:
+        raise
+
+    config_tester.check_channel()
+    config_tester.check_provider(disable_launcher=True)
+    config_tester.check_provider(disable_launcher=False)
+    config_tester.check_executor(square, 3, 9)
```

Comment:
Maybe wrap these four checkes into a `check` function, and you can avoid calling all these functions twice in `cli_run()` and `check_config`